.github/workflows/on-push.yml

```diff
@@ -8,32 +8,36 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
       - uses: volta-cli/action@v1
-      - name: Cache node_modules
-        id: cache-primes
-        uses: actions/cache@v1
+      - uses: actions/cache@v2
         with:
-          path: ./node_modules
-          key: ${{ runner.os }}-${{hashFiles('**/package-lock.json')}}-node_modules
-      - run: npm install
-      - run: npm run lint
-      - run: npm run build
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - run: volta pin yarn@latest
```

Comment:
I don't think you need to pin it.  You were missing yarn in volta config :)

Reply:
1 way to find out!